ledger/src/bank_forks_utils.rs

```diff
@@ -117,25 +113,25 @@ fn load_from_snapshot(
     process_options: ProcessOptions,
     transaction_status_sender: Option<&TransactionStatusSender>,
     cache_block_meta_sender: Option<&CacheBlockMetaSender>,
-    archive_filename: PathBuf,
-    archive_slot: Slot,
-    archive_hash: Hash,
-    archive_format: ArchiveFormat,
+    snapshot_archive_info: SnapshotArchiveInfo,
```

Comment:
I did some experiments and examine the assembly code generated in a site Jeff W. pointed me to -- it does not appear the compiler is doing the move. I still see for each fields pushing to the stack instead of pushing one pointer and referencing the fields by offset later. But I agree you did not make it worse compared with before. And let's just see if you need copy the struct.

Reply:
OK thanks for checking. I've changed the code to use a reference now.